plugins/base/src/main/kotlin/translators/documentables/documentableLanguage.kt

```diff
@@ -2,14 +2,11 @@ package org.jetbrains.dokka.base.translators.documentables
 
 import org.jetbrains.dokka.DokkaConfiguration
 import org.jetbrains.dokka.analysis.PsiDocumentableSource
+import org.jetbrains.dokka.model.Language
 import org.jetbrains.dokka.model.WithSources
 
-internal enum class DocumentableLanguage {
-    JAVA, KOTLIN
-}
-
-internal fun WithSources.documentableLanguage(sourceSet: DokkaConfiguration.DokkaSourceSet): DocumentableLanguage =
+internal fun WithSources.documentableLanguage(sourceSet: DokkaConfiguration.DokkaSourceSet): Language =
```

Comment:
Why do you not use this function to determine a source language?
It works with `Classlikes`, `DFunction`, `DProperty`.

Reply:
Yes, but it does not work on DParameter, which is where we have been having issues. Specifically, with the receiver types of extension functions.

Reply:
Actually, `withSources` does something similar to what I am doing here: `descriptor.toSourceElement.containingFile.toString()`
Making `DParameter` `withSources` might be a cleaner solution. I'll take a look.

Reply:
Arguably, other plugin developers might need something similar, and having a `HasSourceLanguage` marker interface and some way to easily get the source language (like an extra) would be better than some internal function on `WithSources` that you don't know about

I would consider this to be an API improvement more than solving a specific problem for Dackka :)